src/components/structures/ThreadView.tsx

```diff
@@ -203,6 +204,15 @@ export default class ThreadView extends React.Component<IProps, IState> {
             event_id: this.state.thread?.id,
         };
 
+        let previousPhase = RightPanelStore.getSharedInstance().previousPhase;
+        // Make sure the previous Phase is always one of the two: Timeline or ThreadPanel
+        if (![RightPanelPhases.ThreadPanel, RightPanelPhases.Timeline].includes(previousPhase)) {
```

Comment:
As discussed, we should probably guard this with the widgets feature flag, so there's no functional change for threads work by itself.